core/account-id/src/lib.rs

```diff
@@ -177,23 +113,21 @@ impl AccountId {
     /// use near_account_id::AccountId;
     ///
     /// let near_tla: AccountId = "near".parse().unwrap();
-    /// assert!(near_tla.is_top_level_account_id());
+    /// assert!(near_tla.is_top_level());
     ///
     /// let alice: AccountId = "alice.near".parse().unwrap();
     /// assert!(alice.is_sub_account_of(&near_tla));
     ///
     /// let alice_app: AccountId = "app.alice.near".parse().unwrap();
+    ///
+    /// // While app.alice.near is a sub account of alice.near,
+    /// // app.alice.near is not a sub account of near
     /// assert!(alice_app.is_sub_account_of(&alice));
+    /// assert!(!alice_app.is_sub_account_of(&near_tla));
     /// ```
-    pub fn is_sub_account_of(&self, parent_account: &AccountId) -> bool {
-        if parent_account.len() >= self.len() {
-            return false;
-        }
-        // Will not panic, since valid account id is utf-8 only and the length is checked above.
-        // e.g. when `near` creates `aa.near`, it splits into `aa.` and `near`
-        let (prefix, suffix) = self.0.split_at(self.len() - parent_account.len());
-
-        prefix.find('.') == Some(prefix.len() - 1) && suffix == parent_account.as_ref()
+    pub fn is_sub_account_of(&self, parent: &AccountId) -> bool {
+        self.strip_suffix(&parent[..])
+            .map_or(false, |s| !s.is_empty() && s.find('.') == Some(s.len() - 1))
```

Comment:
```suggestion
        self.strip_suffix(parent.as_str())
            .map_or(false, |s| s.ends_with('.'))
```

Reply:
Well, this actually isn't what we want. `app.alice.near` is not a sub-account of `near`.